http/src/v0/block.rs

```diff
@@ -108,31 +112,176 @@ pub fn put<T: IpfsTypes>(
+            InvalidBoolean(ref k, ref v) => write!(fmt, "field {:?} invalid boolean: {:?}", *k, *v),
+        }
+    }
+}
+
+impl<'a> TryFrom<&'a str> for RmOptions {
+    type Error = ParseError<'a>;
+
+    fn try_from(q: &'a str) -> Result<Self, Self::Error> {
+        use ParseError::*;
+
+        let parse = url::form_urlencoded::parse(q.as_bytes());
+
+        let mut args = Vec::new();
+        let mut force = None;
+        let mut quiet = None;
+
+        for (key, value) in parse {
+            let target = match &*key {
+                "args" => {
+                    args.push(value.into_owned());
+                    continue;
+                }
+                "force" => &mut force,
+                "quiet" => &mut quiet,
+                _ => {
+                    // ignore unknown fields
+                    continue;
+                }
+            };
+
+            // common bool field handling
+            if target.is_none() {
+                match value.parse::<bool>() {
+                    Ok(value) => *target = Some(value),
+                    Err(_) => return Err(InvalidBoolean(key, value)),
+                }
+            } else {
+                return Err(DuplicateField(key));
+            }
+        }
+
+        Ok(RmOptions {
+            args,
+            force: force.unwrap_or(false),
+            quiet: force.unwrap_or(false),
+        })
+    }
+}
+
+async fn rm_block<T: IpfsTypes>(ipfs: Ipfs<T>, cid: Cid) -> Result<RmResponse, Error> {
+    match ipfs.remove_block(&cid).await {
+        Ok(()) => Ok(RmResponse {
+            hash: cid.to_string(),
+            error: "".to_string(),
+        }),
+        Err(e) => Ok(RmResponse {
+            hash: "".to_string(),
+            error: e.to_string(),
+        }),
+    }
+}
+
+async fn rm_blocks<T: IpfsTypes>(
+    ipfs: Ipfs<T>,
+    cids: Vec<Cid>,
+) -> Result<impl Stream<Item = Result<RmResponse, Error>> + Send, Error> {
+    use futures::stream::FuturesOrdered;
+
+    let futs: FuturesOrdered<_> = cids
+        .into_iter()
+        .map(|cid| rm_block(ipfs.clone(), cid))
+        .collect();
+
+    Ok(futs)
+}
+
+async fn rm_query<T: IpfsTypes>(
+    ipfs: Ipfs<T>,
+    options: RmOptions,
+) -> Result<impl Reply, Rejection> {
+    let cids: Vec<Cid> = options
+        .args
+        .into_iter()
+        .map(|hash| Cid::from_str(&hash).map_err(StringError::from).unwrap())
+        .collect();
+
+    println!("{:?}", cids);
+
+    let st = rm_blocks(ipfs, cids)
+        .await
+        .map_err(StringError::from)?
+        .map(|result| result.unwrap())
+        .map(|response| serde_json::to_string(&response))
+        .map(|result| match result {
+            Ok(mut string) => {
+                string.push('\n');
+                Ok(string.into_bytes())
+            }
+            Err(e) => {
+                log::error!("edge serialization failed: {}", e);
+                Err(HandledErr)
+            }
+        });
```

Comment:
Might be worth while to try to later push these back to the StreamResponse and make it a "JsonNewLineResponse" since they are now repeated in 3 places I think.

Reply:
I think it's safe to assume, based on @alanshaw 's https://github.com/alanshaw/iterable-ndjson, that any time we're returning ndjon that it's expected as a stream, like in the case of removing a large amount of blocks, but if worse comes to worst I'll switch to a non-streaming response